pkg/git/repository.go

```diff
@@ -252,6 +257,96 @@ func (o *CloneOptions) GetRepo(ctx context.Context) (Repository, fs.FS, error) {
 	return r, fs.Create(bootstrapFS), nil
 }
 
+var validateRepoWritePermission = func(ctx context.Context, r *repo) error {
+	cert, err := r.auth.GetCertificate()
+	if err != nil {
+		return fmt.Errorf("failed getting repository certificates: %w", err)
+	}
+
+	err = r.Repository.PushContext(ctx, &gg.PushOptions{
+		Auth:     getAuth(r.auth),
+		Progress: r.progress,
+		CABundle: cert,
+	})
+	if err != nil {
+		if !errors.Is(err, gg.NoErrAlreadyUpToDate) {
+			// means there was already a commit to push, and it failed to push it
+			return fmt.Errorf("failed to initialize repository, failed pushing to repository: %w", err)
+		}
+		// no commit to push, all up to date
+		err = r.PushDummyCommit("Validating write permission to project", ctx)
+
+		if err != nil {
+			return fmt.Errorf("failed to initialize repository, failed pushing to repository: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (r *repo) PushDummyCommit(commitMsg string, ctx context.Context) error {
+	gitlog, err := r.Repository.Log(&gg.LogOptions{})
+	if err != nil {
+		return fmt.Errorf("failed getting git log: %w", err)
+	}
+
+	commit, err := gitlog.Next()
+	if err != nil {
+		return fmt.Errorf("gailed getting last commit: %w", err)
+	}
+
+	hash := commit.Hash
+	_, err = r.Persist(ctx, &PushOptions{
+		CommitMsg: commitMsg,
+	})
+
+	if err != nil {
+		return fmt.Errorf("failed pushing commit to repository: %w", err)
+	}
+
+	err = r.Reset(ctx, hash)
+
+	if err != nil {
+		log.G().Warnf("failed to reset commit: %v", err)
+	}
+
+	return nil
+
+}
+func (r *repo) Reset(ctx context.Context, hash plumbing.Hash) error {
+	w, err := worktree(r)
+
+	if err != nil {
+		return fmt.Errorf("failed getting worktree of repository: %w", err)
+	}
+
+	err = w.Reset(&gg.ResetOptions{
+		Commit: hash,
+		Mode:   gg.HardReset,
+	})
+
+	if err != nil {
+		return fmt.Errorf("failed getting reseting validation commit: %w", err)
+	}
+
+	cert, err := r.auth.GetCertificate()
+
+	if err != nil {
+		return fmt.Errorf("failed getting repository certificates: %w", err)
+	}
+	err = r.Repository.PushContext(ctx, &gg.PushOptions{
+		Auth:     getAuth(r.auth),
+		Progress: r.progress,
+		CABundle: cert,
+		Force:    true,
+	})
+
+	if err != nil {
+		return fmt.Errorf("failed pushing reset validation commit: %w", err)
+	}
+
+	return nil
+}
```

Comment:
I would give up all this reset part, I don't think it's that bad to leave one "Dummy" commit, that was used to assert write permissions.
More code = more bugs.
I would rather have a dummy commit left in the repo than to crash because of this part.

Reply:
ok np, @noam-codefresh though user shouldnt see it, but it was not that simple to reset it